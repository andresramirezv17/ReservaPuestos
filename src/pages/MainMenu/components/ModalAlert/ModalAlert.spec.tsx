import React from 'react';
import { ModalAlert, ModalProps } from '.';
import { render, fireEvent } from '@testing-library/react';

describe('ModalAlert test', () => {
  let props: ModalProps;
  beforeEach(() => {
    props = {
      showModal: true,
      Title: 'InformaciÃ³n',
      body: 'Alert example',
      btnCancel: 'Cancelar',
      btnContinue: 'Continuar',
      showAlert: jest.fn(),
    };
  });

  it('should match with snapshot', () => {
    const { container } = render(<ModalAlert {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('Should call showAlert when button is clicked', () => {
    const { getByTestId } = render(<ModalAlert {...props} />);
    const button = getByTestId('cancel');
    fireEvent.click(button);
    expect(props.showAlert).toHaveBeenCalled();
  });
});
