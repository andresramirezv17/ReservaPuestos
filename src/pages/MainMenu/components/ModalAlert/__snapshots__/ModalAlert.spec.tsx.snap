// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAlert test should match with snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
